aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -530,12 +530,20 @@ at::Tensor PackedLinearWeightsOnednn::apply_dynamic_impl(
     x_min = t_min.item<float>();
   }
 #endif
-  const int precision = 8;
+
+  static constexpr int precision = 8;
+#if defined(__aarch64__)
+  // oneDNN+ACL has optimized kernels for s8s8 matmul, so input is signed
+  static constexpr bool is_signed = true;
+#else
+  static constexpr bool is_signed = false;
+#endif
+
   auto q_params = quant_utils::ChooseQuantizationParams(
       /*min=*/x_min,
       /*max=*/x_max,
-      /*qmin=*/0,
-      /*qmax=*/(1 << precision) - 1,
+      /*qmin=*/is_signed ? -(1 << (precision - 1)) : 0,
+      /*qmax=*/is_signed ? ((1 << (precision - 1)) - 1) : (1 << precision) - 1,
```

Comment:
Why not alias `qtype` to signed or unsigned int8 type and than just use numeric_limits to fetch the appropriate values, rather than compute it by hand?
```suggestion
      /*qmin=*/std::numeric_limits<qtype>::min(),
      /*qmax=*/std::numeric_limits<qtype>::max(),
```

Reply:
Done, thanks!